src/cockpit/packages.py

```diff
@@ -43,6 +44,29 @@ def directory_items(path):
     return sorted(path.iterdir(), key=lambda item: item.name)
 
 
+def parse_accept_language(accept_language: str) -> List[Tuple[str, str]]:
+    """Parse the Accept-Language header
+
+    Returns an ordered list of (language, weight) pairs.
+
+    https://tools.ietf.org/html/rfc7231#section-5.3.5
+    """
+
+    accept_languages = accept_language.split(',')
+    locales = []
+    for index, language in enumerate(accept_languages):
+        if ';' in language:
+            locale, weight = language.split(';')
+            locale = locale.strip()
+            weight = weight.split('=')[-1].strip()
+            locales.append((locale, weight))
+        else:
+            # Retain given locales order by assiging a fake weight
+            locales.append((language.strip(), str(1 - (index / 100))))
+
+    return list(reversed(sorted(locales, key=lambda k: k[1])))
```

Comment:
`sorted` takes a `reverse=` kwarg.  Your use of `reversed()` here reverses the original order, but `reverse=` would preserve it, removing the need to do the indexing trick above.

I also don't see the point of returning the tuples ­— you really just care about the ordered list.

Reply:
This function could return empty strings as locales.  It might be worth trying to defend against that.

Reply:
* Sure, can drop the tuples
* empty strings as locales, hmm requires a test?